### ISSUE TITLE
Could pre-commit have an explicit error message when used with a non-compatible git version ?

### ISSUE THREAD

**Issue by Lucas-C** [pre-commit/pre-commit#433]
### Context

A coworker was using `pre-commit` under Cygwin with git version 2.10.2.windows.1 (included in [Git For Windows](//git-for-windows.github.io/)).
He got the following error:
```
An unexpected error has occurred: CalledProcessError: Command: ('/cygdrive/f/devhome/app/Git/bin/git', 'apply', '/home/user_name/.pre-commit/patch1480087146')
Return code: 128
Expected return code: 0
Output: (none)
Errors: 
    fatal: can't open patch '/home/user_name/.pre-commit/patch1480087146': No such file or directory

Traceback (most recent call last):
  File "/home/user_name/.local/lib/python3.4/site-packages/pre_commit/staged_files_only.py", line 44, in staged_files_only
    yield
...
  File "/home/user_name/.local/lib/python3.4/site-packages/pre_commit/util.py", line 189, in cmd_output
    returncode, cmd, retcode, output=(stdout, stderr),
pre_commit.util.CalledProcessError: Command: ('/cygdrive/f/devhome/app/Git/bin/git', 'reset', '3fa02652357ff0dbb42b5bc78c673b7bc105fcf3', '--hard')
Return code: 128
Expected return code: 0
Output: (none)
Errors: 
    fatal: Not a git repository (or any of the parent directories): .git
...
```

### Suggestion

I think we could detect thhis issue and have a more explicit error message, like "Incompatible version of git detected".
What do you think ?

**Comment by asottile** [comment]
Duplicate of https://github.com/pre-commit/pre-commit/issues/354

Though 2c:

I think hiding this error message is potentially less helpful than showing it.  At least with it being shown, the specific error message can be googled.  The duplicate ticket indicates that a warning should be printed which will probably help with triaging this specific error.  The difficult part is detecting the cygwin / non-cygwin mismatch :)

**Comment by asottile** [comment]
I've made this a warning in #436

**Comment by Lucas-C** [comment]
Thanks !

**Comment by asottile** [comment]
This fix is in the (just released) 0.9.4 release.
